### PR TITLE
add RedisService

### PR DESCRIPTION
**Pull request by 123yael**

### CODE REVIEW

File: src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    private String EXCHANGE_RATE_API = "http://dbd132af-315b-45ea-a839-453d97370cf6.mock.pstmn.io/exchangeRate?companyCurrency=%s&orderCurrency=%s";
+
+    @Autowired
+    private RedisService redisService;
+
+    @SneakyThrows
+    public double convertCurrency(String fromCurrency, String toCurrency) {
+        String rate;
+        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java:26]: SimpleDataFormat is old fashion
we should use LocalDateTime.now().format ... 
this not must ..

File: src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        String date = sdf.format(new Date());
+        String key = fromCurrency + toCurrency + date;
+        if (fromCurrency.equals(toCurrency))
+            return 1;
+        if (redisService.isKeyExists(key))
+            rate = redisService.getValue(key);
+        else {
+            String url = String.format(EXCHANGE_RATE_API, fromCurrency, toCurrency);
+            OkHttpClient client = new OkHttpClient().newBuilder().build();
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java:35]: Okhttp is old fashion
we should use WebClient ... 
this not must ..

File: src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Autowired
+    private RedisService redisService;
+
+    @SneakyThrows
+    public double convertCurrency(String fromCurrency, String toCurrency) {
+        String rate;
+        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+        String date = sdf.format(new Date());
+        String key = fromCurrency + toCurrency + date;
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java:28]: Please create object with from,to,date and put it as key

**Review by moriyahazan — CHANGES_REQUESTED** [review]
nice!

File: src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java (hunk 2 above)
Comment (123yael) [reply src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java]: This is the example the postman brought me. I changed

**Comment by 123yael** [comment]
Thanks for all the comments

File: src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java (hunk 1 above)
Comment (123yael) [reply src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java]: It was a temporary thing I forgot to change, thanks

File: src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Value("${exchangeRateApi}")
+    private String exchangeRateApi;
+    @Autowired
+    private RedisService redisService;
+
+    @SneakyThrows
+    public double convertCurrency(String fromCurrency, String toCurrency) {
+        String rate;
+        DailyGateKey dailyGateKey = new DailyGateKey(fromCurrency, toCurrency, LocalDate.now());
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java:25]: I mean to save object as is in redis not convert to String
but also this syntax is good

**Review by moriyahazan — APPROVED** [review]

**Pull request merged**